constraint-solver/src/solver.rs

```diff
@@ -203,13 +229,58 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display + Debug> Solver<T, V>
             false
         }
     }
+
+    /// Given a set of variable assignments, checks if they satisfy all the constraints.
+    /// Note that this might return false positives, because it does not propagate any values.
+    fn check_assignments(&self, assignments: &BTreeMap<V, T>) -> bool {
+        let constraints = self
+            .constraint_system
+            .get_constraints(assignments.keys().cloned());
+
+        for constraint in constraints {
+            match constraint {
+                ConstraintSystemItem::AlgebraicConstraint(index) => {
```

Comment:
Could we keep ConstraintSystemItem private? This `index`-mapping is something I would not like to expose. We could either use a proper enum with references - something I would have used in the first place if it would be possible to reference something inside the same struct.